server/models/executor.js

```diff
@@ -1,14 +1,53 @@
+var assert = require('assert');
 var async = require('async');
+var debug = require('debug')('strong-mesh-models:service-manager');
+var instModelWatcher = require('../../lib/helper').instModelWatcher;
+var shouldWatch = require('../../lib/helper').shouldWatch;
 
 module.exports = function(Executor) {
 
   // Refer to server-service.js for description of instance vs currentInstance
   // vs data.
 
+  var name = 'executor';
+
   Executor.observe('after save', function(ctx, next) {
     var serviceManager = Executor.app.serviceManager;
-    var instance = ctx.instance || ctx.currentInstance;
+    debug('------------------ Executor SAVE %j.', ctx.where);
+
+    if (serviceManager._dbWatcher) {
+      if (!shouldWatch(serviceManager, name)) {
+        setImmediate(next);
+        return;
+      }
+      var watcherCtx = {
+        'modelName': name,
+        'watcher': serviceManager._dbWatcher,
+        'saveFun': saveObserver,
+        'saveNext': next,
```

Comment:
This is still experimental.  In this version, both saveObserser part of the original Executor.observe callback *and* next are done in the dbWatcher event listener.  This version of implementation is experimental because the saveNext is set once and re-used every time which is bogus.